### PR TITLE
Implement Login and Sign Up Page for Application

### PR DESCRIPTION
**Pull request by asdfghjkxd**
Let's move to implement the login and signup page of the application.

### CODE REVIEW

**Review by calistatan03 — APPROVED** [review]
looks good to me!! i like the clean design :) just need to add in the header component that katy has created once her PR is merged (or can merge your PR first and include the component in your next PR too, both ways work), otherwise alls good

**Comment by asdfghjkxd** [comment]
@calistatan03 Thank you! I will merge main into this branch and integrate the Header component with the login/signup pages!

**Comment by asdfghjkxd** [comment]
@KatyayaniPande I moved the Header component from the app/ directory to the /components directory for better management of custom components!

**Comment by calistatan03** [comment]
heyy actually, i realised that having donor login and beneficiary login in the header may be abit unnecessary since we can simply show the login page that you have created and users can just toggle to whichever profile login suits them? i was thinking maybe instead of having login for all 3 profiles in the header, can just do one 'login' button and also include a 'signup' button?

**Comment by asdfghjkxd** [comment]
That makes sense, we don't have to repeat ourselves and we can showcase the signup feature as well!

**Comment by calistatan03** [comment]
Ok I'll just let Katy know first! Also wanted to clarify w u, will you be implementing user authentication? I realised w/o authentication it may be difficult for us to merge downstream tasks w the database w/o knowing the logged-in user's details? For example when a donor submits a donation or when a beneficiary submits a request for food, we don't know their profile...or do you have a workaround for this such that we don't need to implement user auth?

**Comment by asdfghjkxd** [comment]
I'm not very sure how to implement auth at the moment, but I assume we can store the user's token/username in the session state and use it to retrieve data from the database once the user logs in with valid credentials.

Do let me know if there are better ways of doing this!

**Comment by calistatan03** [comment]
Actually, was wondering if you've used Firebase before as they have very easy login and sign up methods...and I think its easier to use than MongoDB...

**Comment by asdfghjkxd** [comment]
@calistatan03 Auth is implemented (its quite shaky at the moment, there's a lot of hoops to jump through to get it to work) but it is largely working(?)

The current user can be obtained by using next-auth's useSession() provider, which should be useful for the downstream workflows that require the current user.

AI was used to troubleshoot and generate some of the code pieces; I'm not sure if it is up to standards or otherwise. Would really appreciate it if you are able to take a look to see if there are any areas for optimisation/fixing!

File: app/api/auth/[...nextauth]/route.ts
Comment (asdfghjkxd) [inline app/api/auth/[...nextauth]/route.ts:1]: Rename file to [...nextauth].ts.

Refer to: https://next-auth.js.org/getting-started/typescript

**Review by asdfghjkxd — COMMENTED** [review]
TODOs to complete by 10pm, 24 Aug

**Comment by calistatan03** [comment]
got it! thanks :) sorry im out so not able to look through in detail yet, but just to check the user object being stored in session storage will have all the user info (basically the info they key in during sign-up)? not just email and role ?

**Comment by asdfghjkxd** [comment]
Oh so far its only the role and email, but we can expand it to store the other details as well!

**Comment by asdfghjkxd** [comment]
I have expanded the data to the other fields of the user as well.

Screenshot 2024-08-24 at 9 29 23 PM

**Comment by asdfghjkxd** [comment]
The deployment to Vercel is currently failing due to the presence of a duplicate tailwind.config.js file. This PR will remove the file, which should solve the issue.

**Pull request merged**